### ISSUE TITLE
Print multi page PDF on Android and iOS

### ISSUE THREAD

**Issue by alaindeseine** [CodeMinion/another_brother#89]
Hi @CodeMinion, 

I'm back with a problem!

I need to print a pdf which has more than one page. like this one: 
[1d03fb0559d2ea414911ab699d15a4fb.pdf](https://github.com/user-attachments/files/16015571/1d03fb0559d2ea414911ab699d15a4fb.pdf)

The print `printPdfFile` function prototype permit to indicate a page number (int) to print.

When i check the Brother SDK documentation, i see that (for Android, i have not yet see for iOs) the `printPDF` function has 3 prototypes: 

- public PrintError printPDF(final String path, final PrintSettings printSettings);
- public PrintError printPDF(final String[] paths, final PrintSettings printSettings);
- public PrintError printPDF(final String path, final int[] pages, final PrintSettings printSettings);

The first prototype permit to print all the pages of a PDF file.

When i look at your implementation it seems that you don't implement it. Am i right ?

Do you think it is possible to implement it ? or Not ?

Best regards

Alain Deseine.

**Comment by CodeMinion** [comment]
Hi @alaindeseine ,

Thanks for reaching out about this. You are right, I just looked at the implementation as well and that path is not implemented. 

I'll set some time aside this weekend and add it and let you know once the new version is out. 

Thanks again for reaching out about this and letting me know of the missing implementation,

**Comment by alaindeseine** [comment]
Hi @CodeMinion 
Thanks a lot!
Tell me if i can help.
Regards.

**Comment by SoCaliOS** [comment]
I add the following and it works:

      // Loop through each page and print it
      int totalPageCount = pageCount ?? 1;
      for (int pageNumber = 1; pageNumber <= totalPageCount; pageNumber++) {
        brother.PrinterStatus printResult = await printer.printPdfFile(filepath, pageNumber);
        await processPrinting(printResult, context);
      }

**Comment by alaindeseine** [comment]
@SoCaliOS 

You're right, but there is an API in Brother's SDK that handle printing of all PDF pages. Using it, in my case, will save me from opening the PDF to count the pages. 

My application operates in production conditions in the food industry and any second saved is precious.

**Comment by CodeMinion** [comment]
Hi @alaindeseine ,

Apologies for the delay. Unfortunately on a second pass I noticed that the APIs mentioned are part of version 4 of the Brother mobile SDK while another_brother was built on top of version 3. The two versions use very different ways of printing so currently until we do a major rewrite of the Android implementation the only way to accomplish the multi page print is by using the suggestion from @SoCaliOS.

**Comment by alaindeseine** [comment]
Hi @CodeMinion 

I understand. 

Have you already planned to upgrade to V4 or not at all?

I had already practiced as @SoCaliOS  suggests to get around the problem earlier last June. But I preferred to wait for the modification before using this workaround.

So I'm going to implement it while waiting for the transition to V4.

Thank you again for your support.

Best regards,
